woof-code/rootfs-skeleton/usr/sbin/dotpup

```diff
@@ -51,12 +51,12 @@ BOOTFLASHBUTTON=$(install_button \
 	"$(gettext "Quickly create a bootable usb flash drive")" \
 	'bootflash &' flashcard_usb.svg )
 
-if [ -d /usr/share/g4dos -o -f /usr/local/frugalpup/grub4dos.tar.xz -o -f /usr/lib/grub4dos/grldr ] ; then
+#if [ -d /usr/share/g4dos -o -f /usr/local/frugalpup/grub4dos.tar.xz -o -f /usr/lib/grub4dos/grldr ] ; then
 	FULLINSTALLBUTTON=$(install_button \
 	"$(gettext "Installer")" \
-	"$(gettext "Currently doesn't support UEFI")" \
+	"$(gettext "Hard Drive Install:")" \
```

Comment:
This label was true, `puppyinstaller` doesn't support UEFI.

Reply:
No, the hard drive installer works with uefi, I have uefi and have tested it. Full install (on hard drive) doesn't work with uefi. Also the commit blocking out the grub for dogs was an error ment to leave a commitment that I need to come back and improve, because it still needs improvement, it only checks for grub 4dos. It does not check if grub 2 is already installed by another os ...

Reply:
To be more clear the puppy installer works with uefi in frugal mode. It does not work with "full install".. for more context that you probably not aware of ( this is some stuff from WAY. Back.. ) puppy Linux used to have a install method that would un layer the file system and install it like a " normal" Linux distro to a hard drive .. this full install method is the one that doesn't work with uefi. Full install was never recommended, because it takes away all the advantages of puppy. It was only useful in a handful of situations...

Reply:
Look closely at puppyinstaller, it refuses to do anything if you have GPT, and it never mounts the ESP, so it can't put the kernel there. I think your machine is configured to allow BIOS boot and what's why the installer "works" with UEFI.

Reply:
Nope it's uefi. I mean I can send a full boot video if you don't believe me. The only 2 places I have still had issues was with my $200 2 in 1 that only has flash memory, and inside Virtual box ( virtual box only supports uefi with certain os, so I didnt do any more looking into it). I want to say the drive I have fossfa64 installed to is has a gpt table ( but I could be wrong here.) . I have spent  40 + hours working on this, I have probably done 8 plus install on 2 different machines.. one of those installs I tried a traditional full install ( just to see what would happen) it didn't boot I'm presuming because of the uefi.. I don't have the full specs of my laptop on me but my gaming PC specs  as follows: 
Mobo ASRock b550 gaming
32 gbs ram
Ryzen 5600g
Rx6600 gpu
I have 3 storage device 
1tb nvme
4tb spinning disk
500 gb SSD - with fosfa installed here on a 125 gb partition.

So the installer is still a little wonky, it Dosent tuch the boot partition it calls grub4 dos at the end, if you have grub4 dos then it does scan the system and suggest a boot menu lay out ( my laptop has only grub 4 dos) it dosent work as well with grub2, you have to manually set the menu  it see 
https://forum.puppylinux.com/viewtopic.php?t=437 for reference.. 
For a good "how too" but the installer does do a pop up with the relevant info. 
Generally speaking for most Linux distros after install you would go back to your main distro and just update grub, grub would then find you the Linux distro and add it to the menu, grub 2 dosen't  find puppy correctly....  ( I plan on trying to find fixes for this later but it's a grub2 problem I believe. ) 

But to sum it up unless my asrock 550b came factory set to legacy boot, and the full installer is broken elsewhere ( somewhat likely) and the warning is getting tripped even with my efi being disabled... It works with uefi, I can triple check this tomorrow and even send a full boot videos with me choosing efi on my laptops one time boot menu if you don't believe me but I mean all three would have had to happen.. so yes the installer works as is one uefi, does it work well. No but it does work..  like I said I could see where your confusion comes from, but please have a look at my changes and give them a try, it works pending it's not a flash memory internal  storage device, and it's not inside Virtual box ( I did not test other virtual systems .. ).

Reply:
To add the issue with the two in one is the installer finding the drive..